src/ring_theory/power_series/basic.lean

```diff
@@ -1283,6 +1283,14 @@ begin
   { intro h, simpa using congr_arg (coeff R 1) h }
 end
 
+lemma rescale_inj {a : R} (ha : a ≠ 0) : function.injective (rescale a) :=
+begin
+  rw function.injective,
+  rintros p q h, rw power_series.ext_iff at *, rintros n, specialize h n,
+  rw [coeff_rescale, coeff_rescale] at h, rw mul_eq_mul_left_iff at h, cases h,
+  { assumption, }, { exfalso, apply ha, apply pow_eq_zero h, },
+end
```

Comment:
```suggestion
lemma rescale_injective {a : R} (ha : a ≠ 0) : function.injective (rescale a) :=
begin
  intros p q h,
  rw power_series.ext_iff at *,
  intros n,
  specialize h n,
  rw [coeff_rescale, coeff_rescale] at h,
  rw mul_eq_mul_left_iff at h,
  cases h,
  { assumption, },
  { exfalso, apply ha, apply pow_eq_zero h, },
end
```
Can the two `rw at h` steps be merged?
The last three lines can probably be something like
```
  apply h.resolve_left,
  intro h',
  exact ha (pow_eq_zero h')
```

Reply:
yes, that works (but with `h.resolve_right`).